packages/react/src/radio-group/RadioGroup.tsx

```diff
@@ -0,0 +1,58 @@
+import * as RadixRadio from "@radix-ui/react-radio-group";
+import {
+  Children,
+  type ComponentPropsWithRef,
+  type ReactElement,
+  cloneElement,
+  forwardRef,
+  isValidElement,
+} from "react";
+
+import type { BoxProps } from "../box";
+
+import { Flex } from "../flex";
+import { RadioGroupItem } from "../radio-group-item";
+import { extractSprinkles } from "../sprinkles";
+
+type RadioGroupProps = BoxProps<
+  typeof RadixRadio.RadioGroup,
+  {
+    children:
+      | ReactElement<typeof RadioGroupItem>
+      | ReactElement<typeof RadioGroupItem>[];
+    readonly?: boolean;
+  }
+>;
+
+export const RadioGroup = forwardRef<HTMLDivElement, RadioGroupProps>(
+  ({ children, disabled, readonly, ...props }, ref) => {
+    const { restProps, sprinkleProps } = extractSprinkles(props);
+
+    return (
+      <Flex asChild gap="sm" {...sprinkleProps}>
+        <RadixRadio.RadioGroup
+          disabled={disabled || readonly}
+          ref={ref}
+          {...restProps}
+        >
+          {Children.map(children, (child) => {
+            if (
+              isValidElement<ComponentPropsWithRef<typeof RadioGroupItem>>(
+                child,
+              )
+            ) {
+              return cloneElement(child, {
+                ...child.props,
+                disabled: child.props?.disabled || disabled,
+                readonly: readonly,
+              });
+            }
+            return child;
+          })}
```

Comment:
we shouldn't do this - there is no guarantee that the children will be available in a flat manner like this - they could be wrapped in fragments - they might be nested deeper inside other components

instead we can re-use the state that radix exposes inside each item and just render the children as-is
```suggestion
          {children}
```